thrust/system/cuda/detail/reduce.inl

```diff
@@ -27,6 +27,7 @@
 #include <thrust/system/cuda/detail/decomposition.h>
 #include <thrust/system/cuda/detail/execution_policy.h>
 #include <thrust/system/cuda/detail/execute_on_stream.h>
+#include <thrust/system/cuda/detail/synchronize.h>
```

Comment:
I think this header is superfluous because `reduce` does not call `synchronize`.